packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextShadowNode.java

```diff
@@ -73,90 +200,10 @@ public long measure(
-                text.setSpan(
-                    new ReactAbsoluteSizeSpan(
-                        (int) Math.max((span.getSize() * ratio), minimumFontSize)),
-                    text.getSpanStart(span),
-                    text.getSpanEnd(span),
-                    text.getSpanFlags(span));
-                text.removeSpan(span);
-              }
-              layout = measureSpannedText(text, width, widthMode);
-            }
-          }
-
-          if (mShouldNotifyOnTextLayout) {
-            ThemedReactContext themedReactContext = getThemedContext();
-            WritableArray lines =
-                FontMetricsUtil.getFontMetrics(
-                    text, layout, sTextPaintInstance, themedReactContext);
-            WritableMap event = Arguments.createMap();
-            event.putArray("lines", lines);
-            if (themedReactContext.hasActiveReactInstance()) {
-              themedReactContext
-                  .getJSModule(RCTEventEmitter.class)
-                  .receiveEvent(getReactTag(), "topTextLayout", event);
-            } else {
-              ReactSoftExceptionLogger.logSoftException(
-                  "ReactTextShadowNode",
-                  new ReactNoCrashSoftException("Cannot get RCTEventEmitter, no CatalystInstance"));
-            }
-          }
-
-          final int lineCount =
-              mNumberOfLines == UNSET
-                  ? layout.getLineCount()
-                  : Math.min(mNumberOfLines, layout.getLineCount());
-
-          // Instead of using `layout.getWidth()` (which may yield a significantly larger width for
-          // text that is wrapping), compute width using the longest line.
-          float layoutWidth = 0;
-          if (widthMode == YogaMeasureMode.EXACTLY) {
-            layoutWidth = width;
-          } else {
-            for (int lineIndex = 0; lineIndex < lineCount; lineIndex++) {
-              float lineWidth = layout.getLineWidth(lineIndex);
-              if (lineWidth > layoutWidth) {
-                layoutWidth = lineWidth;
-              }
-            }
-            if (widthMode == YogaMeasureMode.AT_MOST && layoutWidth > width) {
-              layoutWidth = width;
-            }
-          }
-
-          if (android.os.Build.VERSION.SDK_INT > android.os.Build.VERSION_CODES.Q) {
-            layoutWidth = (float) Math.ceil(layoutWidth);
-          }
-          float layoutHeight = height;
-          if (heightMode != YogaMeasureMode.EXACTLY) {
-            layoutHeight = layout.getLineBottom(lineCount - 1);
-            if (heightMode == YogaMeasureMode.AT_MOST && layoutHeight > height) {
-              layoutHeight = height;
-            }
-          }
-
-          return YogaMeasureOutput.make(layoutWidth, layoutHeight);
+          TextView textView =
+            Assertions.assertNotNull(mInternalView, "mInternalView cannot be null");
+
+          return measureWithView(text, textView, width, widthMode, height, heightMode);
```

Comment:
Yoga measures on a background thread. It's unsafe to access TextView here.

Reply:
I'm sorry, could you offer some points exactly about the unsafe access? If it means some cases like #17530 , I think they are different. Because EditText has a background  by default while TextView doesn't.
```xml
    <style name="Widget.EditText">
        <item name="focusable">true</item>
        <item name="focusableInTouchMode">true</item>
        <item name="clickable">true</item>
        <item name="background">?attr/editTextBackground</item>
        <item name="textAppearance">?attr/textAppearanceMediumInverse</item>
        <item name="textColor">?attr/editTextColor</item>
        <item name="gravity">center_vertical</item>
        <item name="breakStrategy">simple</item>
        <item name="hyphenationFrequency">@dimen/config_preferredHyphenationFrequency</item>
        <item name="defaultFocusHighlightEnabled">false</item>
    </style>
```
```xml
    <style name="Widget.TextView">
        <item name="textAppearance">?attr/textAppearanceSmall</item>
        <item name="textSelectHandleLeft">?attr/textSelectHandleLeft</item>
        <item name="textSelectHandleRight">?attr/textSelectHandleRight</item>
        <item name="textSelectHandle">?attr/textSelectHandle</item>
        <item name="textEditPasteWindowLayout">?attr/textEditPasteWindowLayout</item>
        <item name="textEditNoPasteWindowLayout">?attr/textEditNoPasteWindowLayout</item>
        <item name="textEditSidePasteWindowLayout">?attr/textEditSidePasteWindowLayout</item>
        <item name="textEditSideNoPasteWindowLayout">?attr/textEditSideNoPasteWindowLayout</item>
        <item name="textEditSuggestionItemLayout">?attr/textEditSuggestionItemLayout</item>
        <item name="textEditSuggestionContainerLayout">?attr/textEditSuggestionContainerLayout</item>
        <item name="textEditSuggestionHighlightStyle">?attr/textEditSuggestionHighlightStyle</item>
        <item name="textCursorDrawable">?attr/textCursorDrawable</item>
        <item name="breakStrategy">high_quality</item>
        <item name="hyphenationFrequency">@dimen/config_preferredHyphenationFrequency</item>
    </style>
```

Moreover, There are many cases where `YogaMeasureFunction` measures in the same way like `ReactSwitch`,  `ProgressBar`.

Reply:
I think the change is that with this PR, the ShadowNode now manages a new `TextView` on the background thread that is only used during layout?

Reply:
I'm seeing if we can dig up the context on why Fabric chose to avoid this style of measurement, but this is a pretty major change. Usually we'd want to A/B test something like this, but that isn't really a possibility because our internal apps do not run Paper-specific code. This also means we receive very limited test coverage, since this code is not excercised by tests against our own apps.

If it is possible to find the state we are missing in the layout, that would likely be more palatable. Do you know if this issue reproduces in Fabric?

Reply:
@ jcdhlzq Ah, it seems like creating UI components on a background thread is not as bad on Android as it is on iOS. It seems there's lots of potential gotchas though - https://www.markhansen.co.nz/android-view-thread/

Reply:
I've had A/B test on various devices with different ROMs and put it on PRT environment and No any problem or bad case feedback reported.
As for Fabric mode, we don't use that architecture yet.

Reply:
> @NickGerleman I'm also pretty concerned about memory here, as it means we now have two TextViews for every Text element on screen.

Later, maybe after a few days, I'll have a test about peak memory difference and explain the root cause for this phenomenon which is related to Paint.nGetRunAdvance, a native method which may be modified by different ROM vendors. For recent days, I am too busy to take care about this PR in time. Sorry about that!

Reply:
> @NickGerleman I'm also pretty concerned about memory here, as it means we now have two TextViews for every Text element on screen.

The memory diff is about 1.2kb when measuring with TextView.
![image](https://github.com/facebook/react-native/assets/23273745/9855c475-efa2-49a8-a5e3-7f758693e410)

Reply:
> I'm seeing if we can dig up the context on why Fabric chose to avoid this style of measurement, but this is a pretty major change. Usually we'd want to A/B test something like this, but that isn't really a possibility because most of our internal apps do not run Paper-specific code. This also means we receive very limited test coverage, since this code is not exercised by tests against most of our own apps.
> 
> If it is possible to find the state we are missing in the layout, that would likely be more palatable. Do you know if this issue reproduces in Fabric?

Yes, the issue reproduces when running on the app built with `newArchEnabled=true`.
![image](https://github.com/facebook/react-native/assets/23273745/60926332-5b6c-4fe6-ab24-6ec4eb72b5ae)
![image](https://github.com/facebook/react-native/assets/23273745/305c6f4f-4313-4af8-9f91-ebc820ca246a)